pkg/monitoring/redisreplication.go

```diff
@@ -0,0 +1,122 @@
+package monitoring
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+type MetricDescription struct {
```

Comment:
Don't make this struct directly make the Gauge counter and register it from the main.go

Reply:
> Don't make this struct directly make the Gauge counter and register it from the main.go

The register is already in the init function of main.go

Reply:
This struct seems not much use can we have only vars for the gauge metrics

Reply:
I added changes, to make the struct relevant